File: gameplay/js/model/messageLine.js
catan = catan || {};
catan.models = catan.models || {};

/**
	This module contains the MessageLine

	@module catan.models
	@namepsace models
*/

catan.models.MessageLine = (function() {

  /**
    The MessageLine class is an object containing a single message

    <pre>
      Invariants:
        INVARIANT: the message should never change after it is instantiated
        INVARIANT: the source should never change after it is instantiated

      PRE: message is a string
      PRE: source is a string
      POST: message is now set
      POST: source is now set
    </pre>

    @class MessageLine
    @constructor
    @param {String} the message
    @param {String} the source of the message
    */
  function MessageLine(message, source) { };

  /**
    Returns the message

    <pre>
      PRE: none
      POST: none
    </pre>

    @method getMessage
    @return {String}
   */
  MessageLine.prototype.getMessage = function() { };

  /**
    Returns the source

    <pre>
      PRE: none
      POST: none
    </pre>

    @method getSource
    @return {String}
   */
  MessageLine.prototype.getSource = function() { };

});

File: src/com/catan/datamodel/MessageLine.java
package com.catan.datamodel;

public class MessageLine {
	
	// backing fields
	private String _source;
	private String _message;
	
	// getters and setters
	public String get_source() {
		return _source;
	}
	public void set_source(String _source) {
		this._source = _source;
	}
	public String get_message() {
		return _message;
	}
	public void set_message(String _message) {
		this._message = _message;
	}
}